File: taxiFrontEnd/taxi-f/src/app/entity/Trip.ts
import {UserTaxi} from './UserTaxi';

export class Trip {
  trip_id: number;
  trip_adress_a: string;
  trip_adress_b: string;
  trip_price: number;
  trip_feedback: string;
  trip_user: UserTaxi;
  trip_dirver: UserTaxi;
  trip_status: string;


  constructor(trip_id: number, trip_adress_a: string, trip_adress_b: string, price: number, trip_feedback: string, trip_user: UserTaxi, trip_dirver: UserTaxi, trip_status: string) {
    this.trip_id = trip_id;
    this.trip_adress_a = trip_adress_a;
    this.trip_adress_b = trip_adress_b;
    this.trip_price = price;
    this.trip_feedback = trip_feedback;
    this.trip_user = trip_user;
    this.trip_dirver = trip_dirver;
    this.trip_status = trip_status;
  }
}

File: taxiBeckEnd/src/main/java/app/entity/Trip.java
package app.entity;

import javax.persistence.*;

@Entity
@Table
public class Trip {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long trip_id;
    private String trip_adress_a;
    private String trip_adress_b;
    private int trip_price;
    private String trip_feedback;

    public String getTrip_status() {
        return trip_status;
    }

    public void setTrip_status(String trip_status) {
        this.trip_status = trip_status;
    }

    private String trip_status;
    @ManyToOne
    @JoinColumn(name = "user_id_u")
    private UserTaxi trip_user;
    @ManyToOne
    @JoinColumn(name = "user_id_d")
    private UserTaxi trip_dirver;

    public Trip() {
    }

    public Long getTrip_id() {
        return trip_id;
    }

    public void setTrip_id(Long trip_id) {
        this.trip_id = trip_id;
    }

    public String getTrip_adress_a() {
        return trip_adress_a;
    }

    public void setTrip_adress_a(String trip_adress_a) {
        this.trip_adress_a = trip_adress_a;
    }

    public String getTrip_adress_b() {
        return trip_adress_b;
    }

    public void setTrip_adress_b(String trip_adress_b) {
        this.trip_adress_b = trip_adress_b;
    }

    public int getTrip_price() {
        return trip_price;
    }

    public void setTrip_price(int price) {
        this.trip_price = price;
    }

    public String getTrip_feedback() {
        return trip_feedback;
    }

    public void setTrip_feedback(String trip_feedback) {
        this.trip_feedback = trip_feedback;
    }

    public UserTaxi getTrip_user() {
        return trip_user;
    }

    public void setTrip_user(UserTaxi trip_user) {
        this.trip_user = trip_user;
    }

    public UserTaxi getTrip_dirver() {
        return trip_dirver;
    }

    public void setTrip_dirver(UserTaxi trip_dirver) {
        this.trip_dirver = trip_dirver;
    }
}
